Fix incorrect HA address in generated config

steven-chien opened the pull request:
The prefix of the HA endpoint address should have the same IP as the local endpoint address. This patch merges the two entries.
@andriytk

steven-chien committed:
Fix incorrect hostname in sagercf and list all m0_client addresses

Signed-off-by: steven <wdchien@kth.se>

steven-chien committed:
Merge branch 'Seagate:main' into main

steven-chien committed:
fix incorrect HA address

- The IP address of HA address should be the same as the local end point address

Signed-off-by: steven <wdchien@kth.se>

steven-chien committed:
Merge branch 'main' of https://github.com/KTH-HPC/cortx-motr-apps into main

andriytk commented on scripts/motraddr.sh:
```diff
@@ -43,9 +43,11 @@ p[1]=$(grep -A2 Profile $hastatus | tail -n1 | awk '{print $1}')
 p[2]=$(sed -n '/pools:/,/^[A-Za-z]/p' $hastatus | grep -E 'tier.+p1' | awk '{print $1}')
 p[3]=$(sed -n '/pools:/,/^[A-Za-z]/p' $hastatus | grep -E 'tier.+p2' | awk '{print $1}')
 p[4]=$(sed -n '/pools:/,/^[A-Za-z]/p' $hastatus | grep -E 'tier.+p3' | awk '{print $1}')
+p[5]=$(grep -A$((2+($r)%16)) $c $hastatus | tail -n1 | awk '{print $4}')
```
Isn't it overwritten below at line 53?

steven-chien replied:
Fixed in 5397c7c

andriytk commented on scripts/motraddr.sh:
```diff
@@ -167,7 +169,7 @@ echo "# Application: All"
 
 echo
-echo "HA_ENDPOINT_ADDR = ${p[0]}"
+echo "HA_ENDPOINT_ADDR = $(echo "${p[5]}" | cut -d'@' -f 1)@$(echo "${p[0]}" | cut -d'@' -f 2)"
```
You can use bash's build-in Parameter Expansion feature to split variable value to parts. For example:

```bash
$ aa=bb@cc
$ echo ${aa%@*}@${aa#*@}
bb@cc
```

steven-chien replied:
Changed to use expansion in 5397c7c

andriytk reviewed (changes requested):
The code needs some improvements.

steven-chien committed:
improvements

- remove duplicated local address extraction
- use shell parameter expansion to combine HA address

Signed-off-by: steven <wdchien@kth.se>

andriytk reviewed (changes requested):
Should we drop the code at line 33 for HA_ENDPOINT_ADDR or should we move it down?

nezzzu commented:
@steven-chien I will merge once comments from @andriytk are addressed.

nezzzu commented:
By the way, what's wrong with using p[0] directly for HA address? It's already extracted and available.

shailesh-vaidya commented:
Closing as an obsolete
